resources/lib/common/kodi_library_ops.py

```diff
@@ -68,18 +68,20 @@ def get_library_item_details(dbtype, itemid):
     return json_rpc(method, params)[dbtype + 'details']
 
 
-def scan_library(path=""):
+def scan_library(path=''):
     """Start a Kodi library scanning in a specified folder to find new items"""
     method = 'VideoLibrary.Scan'
-    params = {'directory': path}
+    params = {'directory': makeLegalFilename(translatePath(path))}
     return json_rpc(method, params)
 
 
-def clean_library(show_dialog=True):
+def clean_library(show_dialog=True, path=''):
     """Start a Kodi library scanning to remove non-existing items"""
     method = 'VideoLibrary.Clean'
     params = {'content': 'video',
               'showdialogs': show_dialog}
+    if not G.KODI_VERSION.is_major_ver('18') and path:
```

Comment:
There may still be some people on older versions, maybe use `G.KODI_VERSION.is_less_version`?